test/integration/admin/api/accounts_controller_test.rb

```diff
@@ -4,6 +4,30 @@
 
 class Admin::API::AccountsControllerTest < ActionDispatch::IntegrationTest
 
+  class MasterAccount < ActionDispatch::IntegrationTest
+
+    def setup
+      host! master_account.admin_domain
+    end
+
+    def test_find
+      account = FactoryGirl.create(:simple_provider, provider: master_account)
+      service = FactoryGirl.create(:simple_service, account: account)
+      service.service_tokens.create!(value: 'token')
+
+      get find_admin_api_accounts_path(format: :xml, provider_key: master_account.api_key, buyer_service_token: 'token')
```

Comment:
Would `buyer_service_token: 'token'`also fail?

Reply:
yes, the result is not found.

Reply:
Why would you add `toke` in particular @josemigallas ? The next one is doing it with `123` which is kinda the same to me 😕 what's the difference?

Reply:
For what I read, `get find_admin_api_accounts_path` could return an array of accounts or just one so I imagined it could be like searching by names: `find_by_name('jose') -> ['jose', 'josemi', ..]`

Reply:
noupe, it returns only one `account`, not array.